src/scripts/main.js

```diff
@@ -1,3 +1,258 @@
+
+  for (const title of HEADERS) {
+    const label = document.createElement('label');
+
+    label.innerText = `${title[0].toUpperCase() + title.slice(1)}:`;
+
+    if (title === 'office') {
+      label.appendChild(createSelectBox(title));
+      form.appendChild(label);
+      continue;
+    }
+
+    label.appendChild(createInput(title));
+    form.appendChild(label);
+  }
+
+  form.appendChild(createButton());
+  body.appendChild(form);
+}
+
+const pushNotification = (title, description, type) => {
+  const table = document.querySelector('table');
+
+  const notification = document.createElement('div');
+
+  notification.setAttribute('data-qa', 'notification');
+  notification.className = `notification ${type}`;
+
+  notification.innerHTML = `
+    <h2 class="title">${title}</h2>
+    <p>${description}</p>
+  `;
+  table.insertAdjacentElement('afterend', notification);
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, DELAY_NOTIFICATION);
+};
+
+function isEmployeeDataValid(newName, age) {
+  if (newName.length < 4) {
+    pushNotification('Oops!', 'Name has less than 4 letters', 'error');
+
+    throw new Error('Name has less than 4 letters');
+  }
+
+  if (+age <= 18 || +age > 90) {
+    pushNotification('Oops!', 'Age is less than 18 or more than 90', 'error');
+
+    throw new Error('Age is less than 18 or more than 90')
+  }
+
+  return true;
+}
+
+document.addEventListener('DOMContentLoaded', () => {
+  const itemCollection = document.querySelectorAll('tbody tr');
+  const tableHeader = document.querySelector('thead tr');
+  const tableBody = document.querySelector('tbody');
+
+  const items = mapTableRowsToObjects(itemCollection);
+
+  tableHeader.addEventListener('click', (e) => {
+    const title = e.target.closest('th');
+
+    if (!title) {
+      return;
+    }
+
+    const targetValue = title.innerText.toLowerCase();
+
+    if (!title.hasAttribute('data-order') || title.dataset.order === DESC) {
+      title.setAttribute('data-order', ASC);
+    } else {
+      title.setAttribute('data-order', DESC);
+    }
+
+    const order = title.dataset.order;
+
+    sortItems(items, targetValue, order);
+    rebuildList(items);
+  });
+
+  tableBody.addEventListener('click', (e) => {
+    const activeRow = e.target.closest('tr');
+
+    if (!activeRow) {
+      return;
+    }
+
+    itemCollection.forEach((item) => {
+      item.classList.remove('active');
+    });
+
+    activeRow.classList.add('active');
+  });
+
+  addForm();
+
+  if (document.querySelector('.new-employee-form')) {
+    const form = document.querySelector('.new-employee-form');
+
+    form.addEventListener('submit', (e) => {
+      e.preventDefault();
+
+      const data = new FormData(form);
+      const newName = data.get('name');
+      const age = data.get('age');
+
+      // tranform salary into format '10,000,000,000'
+      const salary = parseInt(data.get('salary')).toLocaleString('en-US');
+
+      if (isEmployeeDataValid(newName, age)) {
```

Comment:
The validation logic here is not correct. The `isEmployeeDataValid` function will throw an error if the data is invalid, which will stop the execution of the code. Instead, you should catch the error and handle it appropriately (e.g., by showing an error message to the user).